Widgets form

leonofre opened the pull request:
Related to #7

leonofre committed:
Added widgets folder

leonofre committed:
Added basic class to widget

leonofre committed:
Added widget form

leonofre committed:
Finished the form method

brenoalvs commented on widgets/partials/admin-monk-language-switcher.php:
```diff
@@ -0,0 +1,27 @@
+	<input class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>" type="text" value="<?php echo esc_attr( $title ); ?>">
+
+	<label><?php _e( esc_attr( 'Show Flags ' ) ); ?></label>
+	<input class="input-checkbox" type="checkbox" name="<?php echo esc_attr( 'view-mode' ); ?>" value="<?php echo esc_attr( 'flag' ); ?>"></br >
```
`esc_attr` function is not needed in hardcoded text as `name` attribute.

brenoalvs commented on widgets/partials/admin-monk-language-switcher.php:
```diff
@@ -0,0 +1,27 @@
+ * @subpackage Monk/Widgets/Partials
+ */
+
+$title = ! empty( $instance['title'] ) ? $instance['title'] : __( 'New title', 'text_domain' );
```
Every localized text using translation functions as `_e` or `__` needs a second parameter called `text-domain`.

Check how these functions works here: https://codex.wordpress.org/I18n_for_WordPress_Developers

You can set all text-domains to `monk` for now.

brenoalvs commented on widgets/partials/admin-monk-language-switcher.php:
```diff
@@ -0,0 +1,27 @@
+	<label for="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>"><?php _e( esc_attr( 'Title: ' ) ); ?></label> 
+	<input class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>" type="text" value="<?php echo esc_attr( $title ); ?>">
+
+	<label><?php _e( esc_attr( 'Show Flags ' ) ); ?></label>
```
Join the `_e` and `esc_attr` functions.
WordPress provides an `esc_attr_e` function to make this, don't forget to set the ext-domain value to `monk`

brenoalvs commented on widgets/class-monk-language-switcher.php:
```diff
@@ -0,0 +1,116 @@
+	public function form( $instance ) {
+
+		// outputs the options form on admin
+		$languages_nat = array(
```
Align array arrows with spaces.

brenoalvs commented on widgets/css/admin-language-switcher-style.css:
```diff
@@ -0,0 +1,13 @@
+* {
```
You'll **NEVER** use CSS global selectors in WordPress plugins.
You can broke all user's themes and plugins doing things like this.

Prefix all plugin classes with `.monk-`
And please follow the CSS code standards.

brenoalvs reviewed (changes requested):
@leonofre please review those changes and also check the code standards.

leonofre committed:
Fixed the previous error

leonofre committed:
Added improvements in the comments

leonofre committed:
Changed permissions on the register_widgets method

leonofre committed:
Added the necessery corrections

leonofre committed:
Fixed new problems in the comments

leonofre committed:
Merge branch 'widgets' of https://github.com/brenoalvs/monk into widgets-form

leonofre committed:
Fixed new problems and moved admin-widgets .css files to admin folder

leonofre committed:
Changed esc_attr_e() method to esc_attr_e() method in label content

Entered attribute name and id as hardcoded

Modified .css file name in admin

Created label class monk-language-switcher-title
